src/javascript/tests/custom_types.rs

```diff
@@ -142,6 +162,26 @@ function update(cat) {
 function access(cat) {
   return cat.cuteness;
 }
+
+function destructure(cat) {
+  let gleam$tmp = cat;
+  if (!(gleam$tmp["type"] === "Cat")) throw new Error("Bad match");
+  let x = gleam$tmp["name"];
+  let y = gleam$tmp["cuteness"];
+  
+  let gleam$tmp = cat;
+  if (!(gleam$tmp["type"] === "Cat")) throw new Error("Bad match");
+  let x$1 = gleam$tmp["name"];
+  
+  let gleam$tmp = cat;
+  if (!(
+    gleam$tmp["type"] === "Cat" &&
+    gleam$tmp["cuteness"] === 4
+  )) throw new Error("Bad match");
+  let x$2 = gleam$tmp["name"];
+  
+  return x$2;
```

Comment:
Can we have this in another text rather than making the examples bigger? I find it makes the test failure output a lot easier to work with if you have small tests that check one thing rather than large ones that check multiple things

Reply:
Fair enough, I was reusing the definition of the Cat type but your right it had got a bit cumbersome